tachyon/math/base/semigroups.h

```diff
@@ -184,6 +184,14 @@ class MultiplicativeSemigroup {
     return ret;
   }
 
+  constexpr auto ExpPowOfTwo(uint32_t log_n) const {
```

Comment:
Though the lines of code that call `ExpPowOfTwo()` do have g = the generator, I agree it would be better to generalize the code. In this case, I'll change the name to `val`, since `self` appears to carry the connotation that `this` would change.